src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -302,6 +302,12 @@ public StorageClientFactory StorageClientFactory
             }
         }
 
+        // Test hook to set storage account. 
+        internal void SetAccountProvider(IStorageAccountProvider provider)
```

Comment:
How do I set the _contextFactory? The setter is private.  Existing tests end up creating a whole new IServiceProvider (and skip JobHostConfiguration altogether) and go down a different path than prod. 
Alternatively - I could make the ContextFactory property internal?